### PR TITLE
Feenk: Announce description property changes

### PR DESCRIPTION
**Pull request by JurajKubelka**
The main purpose is to sync the Feenk repository with the upstream repository.

The main changes:
- add `MATPropertyOwner>>#whenPropertyChangedSend:to:`
- add `MATPropertyOwner>>#unsubscribe:` 
- add `MATPropertyOwner>>#propertyAnnouncerDo:`

Additional changes:
- add announcement notification into `MAActionDescription>>#label:` and
- into `MAActionDescription>>#action:`.

GToolkit `v0.8.3040` uses the description property announcements to dynamically update Magritte widgets. The `GtMagritteModelExamples>>#basicServerObject` example is a basic example that exemplify the usage. The `examples - server` method category of `GtMagritteByScripterExamples` then details how widgets behave. More documentation can be found at https://github.com/feenkcom/gtoolkit/issues/3371#issuecomment-1658840416.

### CODE REVIEW

File: source/Magritte-Deprecated3dot7/MABlockDescription.extension.st
Comment (seandenigris) [inline source/Magritte-Deprecated3dot7/MABlockDescription.extension.st:1]: After you pointed out that this should be an extension method, [it was made one](https://github.com/magritte-metamodel/magritte/commit/d9d2a2f5e7a7fd4d587475962cda7f0cd165d032), but in the Morphic package, not deprecated. See https://github.com/magritte-metamodel/magritte/pull/267/files#r888544985.
Screenshot 2023-08-02 at 9 32 28 PM



Can you remove this from the PR?

File: source/Magritte-Pharo7-Model/MAFileDescription.extension.st
Comment (seandenigris) [inline source/Magritte-Pharo7-Model/MAFileDescription.extension.st:1]: This was moved to Magritte-FileSystem based on your feedback and I think this change is no longer needed. See https://github.com/magritte-metamodel/magritte/pull/267/files#r888549439

File: source/Magritte-Pharo7-PlainPharo/Symbol.extension.st
Comment (seandenigris) [inline source/Magritte-Pharo7-PlainPharo/Symbol.extension.st:1]: A "Magritte-NotGToolkit" package was created for this and other similar issues, so AFAICT this is no longer needed. See https://github.com/magritte-metamodel/magritte/blob/d9d2a2f5e7a7fd4d587475962cda7f0cd165d032/source/Magritte-NotGToolkit/Symbol.extension.st

File: source/Magritte-Pharo7-PlainPharo/package.st
Comment (seandenigris) [inline source/Magritte-Pharo7-PlainPharo/package.st:1]: A "Magritte-NotGToolkit" package was created for this and other similar issues, so AFAICT this is no longer needed. See https://github.com/magritte-metamodel/magritte/tree/d9d2a2f5e7a7fd4d587475962cda7f0cd165d032/source/Magritte-NotGToolkit

File: source/Magritte-Pharo7-PlainPharo/ManifestMagrittePharo7PlainPharo.class.st
Comment (seandenigris) [inline source/Magritte-Pharo7-PlainPharo/ManifestMagrittePharo7PlainPharo.class.st:1]: A "Magritte-NotGToolkit" package was created for this and other similar issues, so AFAICT this is no longer needed. See https://github.com/magritte-metamodel/magritte/tree/d9d2a2f5e7a7fd4d587475962cda7f0cd165d032/source/Magritte-NotGToolkit

**Review by seandenigris — CHANGES_REQUESTED** [review]
The announcement stuff looks interesting. There were a few tangential changes which I believe were already addressed in a different way. See comments...

File: source/Magritte-Pharo7-PlainPharo/Symbol.extension.st
Comment (JurajKubelka) [reply source/Magritte-Pharo7-PlainPharo/Symbol.extension.st:1]: Do you mean that the `Magritte-Pharo7-PlainPharo` package can be removed?

File: source/Magritte-Deprecated3dot7/MABlockDescription.extension.st
Comment (JurajKubelka) [reply source/Magritte-Deprecated3dot7/MABlockDescription.extension.st:1]: Removed.

File: source/Magritte-Pharo7-Model/MAFileDescription.extension.st
Comment (JurajKubelka) [reply source/Magritte-Pharo7-Model/MAFileDescription.extension.st:1]: Removed.

File: source/Magritte-Pharo7-PlainPharo/Symbol.extension.st
Comment (JurajKubelka) [reply source/Magritte-Pharo7-PlainPharo/Symbol.extension.st:1]: Considering the next comment, I guess so. I removed the package.

File: source/Magritte-Pharo7-PlainPharo/package.st
Comment (JurajKubelka) [reply source/Magritte-Pharo7-PlainPharo/package.st:1]: The package is removed.

File: source/Magritte-Pharo7-PlainPharo/ManifestMagrittePharo7PlainPharo.class.st
Comment (JurajKubelka) [reply source/Magritte-Pharo7-PlainPharo/ManifestMagrittePharo7PlainPharo.class.st:1]: The package is removed.

**Review by seandenigris — APPROVED** [review]

**Comment by seandenigris** [comment]
@JurajKubelka I am ready to merge. I was going to squash but I noticed there were 30 commits. Any suggestion whether to squash or not?

**Comment by JurajKubelka** [comment]
@seandenigris: I do not understand what do you mean by "squash" in this context. I believe that we cannot get rid of the 30 commits. What do you think? We can always improve.

**Pull request merged**